src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonPropertyInfo.cs

```diff
@@ -18,7 +18,7 @@ public static partial class DefaultJsonTypeInfoResolverTests
         [Fact]
         public static void JsonPropertyInfoOptionsAreSet()
         {
-            JsonSerializerOptions options = JsonSerializerOptions.Default;
+            JsonSerializerOptions options = new() { TypeInfoResolver = new DefaultJsonTypeInfoResolver() };
```

Comment:
I assume this is because of `Assert.Same(expectedOptions, propertyInfo.Options);` failing? This is because we need to pass in `Options` for original type info rather the one resolved by cache when calling `CreateJsonPropertyInfo`

Reply:
per discussion offline the second part of my statement above is incorrect - we won't actually always go through MakeGenericMethod. I'm not sure if it will be better than before or worse (likely better when there are no recursive types and worse otherwise) but neither of two is perfect so that's ok. The part about Options not having correct reference will likely need to be fixed